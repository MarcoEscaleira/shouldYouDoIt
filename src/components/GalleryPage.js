import React from 'react';
import GalleryImage from './GalleryImage';

const DEFAULT_STATE = {
  doIt: [],
  dontDoIt: []
}

class Gallery extends React.Component {
  state = {
    ...DEFAULT_STATE
  }

  componentWillMount() {
    const requestAll = async () => {
      const response = await fetch('https://shouldyoudoit.herokuapp.com/all');
      const data = await response.json();
      const doIt = data.filter((answer) => answer.msg === 'do it');
      const dontDoIt = data.filter((answer) => answer.msg === 'don\'t do it');
      this.setState(() => ({ doIt, dontDoIt }));
    }
    requestAll();
  }

  render() {
    return (
      <div className="gallery">
        <h1 className="gallery__title">Gallery</h1>

        <div className="gallery__doIt">
          <h1 className="showcase__title">Do It</h1>
          <ul className="gallery__showcase">
            {
              this.state.doIt.map((answer, index) => {
                return (
                  <GalleryImage key={index} answer={answer} />
                )
              })
            }
          </ul>
        </div>
        <div className="gallery__dontDoIt">
          <h1 className="showcase__title">Don't Do It</h1>
          <ul className="gallery__showcase">
            {
              this.state.dontDoIt.map((answer, index) => {
                return (
                  <GalleryImage key={index} answer={answer} />
                )
              })
            }
          </ul>
        </div>
      </div>
    );
  }
}

export default Gallery;