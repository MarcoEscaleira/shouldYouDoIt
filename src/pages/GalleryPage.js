import React, { useState, useEffect } from 'react';
import GalleryImage from '../components/GalleryImage';
import { API_URL } from '../config';

const Gallery = () => {
  const [doIt, setDoIt] = useState([]);
  const [dontDoIt, setDontDoIt] = useState([]);

  useEffect(() => {
    const requestAll = async () => {
      const response = await fetch(`${API_URL}/all`);
      const data = await response.json();
      const doit = data.filter(answer => answer.msg === 'do it');
      const dontdoit = data.filter(answer => answer.msg === "don't do it");

      setDoIt(doit);
      setDontDoIt(dontdoit);
    };
    requestAll();
  }, []);

  return (
    <div className="gallery">
      <h1 className="gallery__title">Gallery</h1>

      <div className="gallery__body">
        <div className="gallery__doIt">
          <h1 className="showcase__title">Do It</h1>
          <ul className="gallery__showcase">
            {doIt.map((answer, index) => (
              <GalleryImage key={index} answer={answer} />
            ))}
          </ul>
        </div>
        <div className="gallery__dontDoIt">
          <h1 className="showcase__title">Don't Do It</h1>
          <ul className="gallery__showcase">
            {dontDoIt.map((answer, index) => (
              <GalleryImage key={index} answer={answer} />
            ))}
          </ul>
        </div>
      </div>
    </div>
  );
};

export default Gallery;
