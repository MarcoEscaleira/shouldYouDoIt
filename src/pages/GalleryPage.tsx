import React, { useState, useEffect } from 'react';
import GalleryImage from '../components/GalleryImage';
import { API_URL } from '../config';
import { DecisionsStateData } from '~store/decisions';

interface RequestedDecisionsStateData extends Array<DecisionsStateData> { }

const Gallery: React.FC = () => {
  const [doIt, setDoIt] = useState<RequestedDecisionsStateData>();
  const [dontDoIt, setDontDoIt] = useState<RequestedDecisionsStateData>();

  useEffect(() => {
    const requestAll = async () => {
      const response = await fetch(`${API_URL}/all`);
      const data: RequestedDecisionsStateData = await response.json();

      const doit: RequestedDecisionsStateData = data.filter(answer => answer.msg === 'do it');
      const dontdoit: RequestedDecisionsStateData = data.filter(answer => answer.msg === "don't do it");

      setDoIt(doit);
      setDontDoIt(dontdoit);
    };
    requestAll();
  }, []);

  return (
    <div className="gallery">
      <h1 className="gallery__title">Gallery</h1>

      <div className="gallery__body">
        <div className="gallery__doIt">
          <h1 className="showcase__title">Do It</h1>
          <ul className="gallery__showcase">
            {doIt?.map((answer, index) => (
              <GalleryImage key={index} answer={answer} />
            ))}
          </ul>
        </div>
        <div className="gallery__dontDoIt">
          <h1 className="showcase__title">Don't Do It</h1>
          <ul className="gallery__showcase">
            {dontDoIt?.map((answer, index) => (
              <GalleryImage key={index} answer={answer} />
            ))}
          </ul>
        </div>
      </div>
    </div>
  );
};

export default Gallery;
