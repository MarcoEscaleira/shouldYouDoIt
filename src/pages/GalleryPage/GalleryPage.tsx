import React, { useState, useEffect } from "react";
import { API_URL } from "~/config";
import { DecisionsStateData } from "~/store/decisions";

import GalleryImage from "~/components/GalleryImage/GalleryImage";
import {
  Container,
  Title,
  Body,
  DoIt,
  DontDoIt,
  SubTitle,
  List
} from "./styled-components";

type RequestedDecisionsStateData = Array<DecisionsStateData>;

const Gallery: React.FC = () => {
  const [doIt, setDoIt] = useState<RequestedDecisionsStateData>();
  const [dontDoIt, setDontDoIt] = useState<RequestedDecisionsStateData>();

  useEffect(() => {
    const requestAll = async () => {
      const response = await fetch(`${API_URL}/all`);
      const data: RequestedDecisionsStateData = await response.json();

      const doit: RequestedDecisionsStateData = data.filter(
        answer => answer.msg === "do it"
      );
      const dontdoit: RequestedDecisionsStateData = data.filter(
        answer => answer.msg === "don't do it"
      );

      setDoIt(doit);
      setDontDoIt(dontdoit);
    };
    requestAll();
  }, []);

  return (
    <Container className="gallery">
      <Title className="gallery__title">Gallery</Title>

      <Body className="gallery__body">
        <DoIt className="gallery__doIt">
          <SubTitle className="showcase__title">Do It</SubTitle>
          <List className="gallery__showcase">
            {doIt?.map((answer, index) => (
              <GalleryImage key={index} answer={answer} />
            ))}
          </List>
        </DoIt>
        <DontDoIt className="gallery__dontDoIt">
          <SubTitle className="showcase__title">Don't Do It</SubTitle>
          <List className="gallery__showcase">
            {dontDoIt?.map((answer, index) => (
              <GalleryImage key={index} answer={answer} />
            ))}
          </List>
        </DontDoIt>
      </Body>
    </Container>
  );
};

export default Gallery;
