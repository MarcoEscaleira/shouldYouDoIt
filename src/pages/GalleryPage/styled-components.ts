import styled from 'styled-components';
import { sizes, pallete } from '../../styles/_settings';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
  margin: 0 0 1rem 0;
`;

export const Title = styled.h1`
  margin: ${sizes.m} 0;
  text-align: center;
  color: ${pallete.sixth_light};
  letter-spacing: 0.2rem;
  font-size: ${sizes.l};
`;

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;

  @media (min-width: ${sizes.desktop_breakpoint}) {
    flex-direction: row;
  }
`;

export const List = styled.ul`
  list-style: none;
  margin: 0px auto;
  padding: 10px;
  max-width: 78rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

export const SubTitle  = styled.h2`
  font-size: 2rem;
  text-align: center;
  color: ${pallete.second_light};
  border-bottom: ${pallete.middle_color} solid 0.1rem;
  padding-bottom: ${sizes.s};
  margin: 0 2rem;
`;

export const DoIt  = styled.div``;

export const DontDoIt  = styled.div``;
