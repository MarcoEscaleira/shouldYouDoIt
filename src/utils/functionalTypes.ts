export type NullaryFn<R> = () => R;
export type UnaryFn<A, R> = (a: A) => R;
export type BinaryFn<A, B, R> = (a: A, b: B) => R;
export type TernaryFn<A, B, C, R> = (a: A, b: B, c: C) => R;
export type QuadFn<A, B, C, D, R> = (a: A, b: B, c: C, d: D) => R;
export type QuinFn<A, B, C, D, F, R> = (a: A, b: B, c: C, d: D, f: F) => R;
export type SixFn<A, B, C, D, F, H, R> = (
  a: A,
  b: B,
  c: C,
  d: D,
  f: F,
  h: H
) => R;
