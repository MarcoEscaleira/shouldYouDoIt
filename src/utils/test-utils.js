import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import store from '../store';

export const renderWithRedux = (ui, initialState) => {
  return {
    ...render(
      <Provider store={store(initialState)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = ui => {
  return {
    ...render(<Router>{ui}</Router>)
  };
};

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
