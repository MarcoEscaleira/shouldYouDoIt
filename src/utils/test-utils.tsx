import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import store, { initialStoreState } from "../store";

export const renderWithRedux = (ui: JSX.Element, initialState?: {}) => {
  const state = {
    ...initialStoreState,
    ...initialState
  };
  return {
    ...render(
      <Provider store={store(state)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = (ui: JSX.Element) => ({
  ...render(<Router>{ui}</Router>)
});

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
