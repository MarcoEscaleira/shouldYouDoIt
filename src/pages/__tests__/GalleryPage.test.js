import React from 'react';
import GalleryPage from '../GalleryPage';
import { renderWithRedux } from '../../utils/test-utils';

describe('Gallery Page test suite', () => {
  it('Should render Gallery Page correctly', () => {
    const { container } = renderWithRedux(<GalleryPage />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
