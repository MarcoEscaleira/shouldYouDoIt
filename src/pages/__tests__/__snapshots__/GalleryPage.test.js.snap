// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Page test suite Should render Gallery Page correctly 1`] = `
<div
  class="gallery"
>
  <h1
    class="gallery__title"
  >
    Gallery
  </h1>
  <div
    class="gallery__body"
  >
    <div
      class="gallery__doIt"
    >
      <h1
        class="showcase__title"
      >
        Do It
      </h1>
      <ul
        class="gallery__showcase"
      />
    </div>
    <div
      class="gallery__dontDoIt"
    >
      <h1
        class="showcase__title"
      >
        Don't Do It
      </h1>
      <ul
        class="gallery__showcase"
      />
    </div>
  </div>
</div>
`;
